merkle/compact/tree.go

```diff
@@ -83,7 +83,9 @@ func NewTreeWithState(hasher hashers.LogHasher, size int64, getNodesFn GetNodesF
 	}
 
 	ids := make([]NodeID, 0, bits.OnesCount64(uint64(size)))
-	for sz := uint64(size); sz != 0; sz &= sz - 1 { // Iterate 1-bits of size.
+	// Iterate over perfect subtrees along the right border of the tree. Those
+	// correspond to 1-bits of the tree size.
```

Comment:
"1-bits of the tree size" isn't an expression that parsed easily to me. From a sample size of 2, 50% of participants found it easy to parse.

Reply:
Done. Does it look clearer?